### PR TITLE
Fixes packet semantics

### PR DESCRIPTION
**Pull request by seunlanlege**
## Issue
*Please replace this line with either a link to the ClickUp issue, or a reference to the GitHub issue.*

## Description
*Please replace this line with info that is not in the ClickUp ticket, eg: "also bumps dependency foo version"*

## Checklist

- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

File: frame/ibc/ibc-primitives/src/lib.rs [hunk 1]
```diff
@@ -28,11 +28,22 @@ pub struct SendPacketData {
...
+	codec::Encode,
+	Default,
+	Debug,
+	codec::Decode,
+	Clone,
+	PartialEq,
+	Eq,
+	Ord,
+	PartialOrd,
+	scale_info::TypeInfo,
```
Comment (blasrodri) [inline frame/ibc/ibc-primitives/src/lib.rs:45]: what's this doing?

File: frame/ibc/src/channel.rs [hunk 2]
```diff
@@ -272,14 +272,16 @@ where
 		let channel_id = key.1.to_string().as_bytes().to_vec();
 		let port_id = key.0.as_bytes().to_vec();
 		let seq = u64::from(key.2);
-		let key = Pallet::<T>::offchain_key(channel_id, port_id);
-		let mut offchain_packets: BTreeMap<u64, OffchainPacketType> =
-			sp_io::offchain::local_storage_get(sp_core::offchain::StorageKind::PERSISTENT, &key)
-				.and_then(|v| codec::Decode::decode(&mut &*v).ok())
-				.unwrap_or_default();
+		// let key = Pallet::<T>::offchain_key(channel_id, port_id);
```
Comment (blasrodri) [inline frame/ibc/src/channel.rs:275]: can't these be removed?

File: frame/ibc/src/lib.rs [hunk 3]
```diff
@@ -243,6 +244,34 @@ pub mod pallet {
 	pub type HostConsensusStates<T: Config> =
 		StorageValue<_, BoundedBTreeMap<u64, IbcConsensusState, ConstU32<250>>, ValueQuery>;
 
+	// temporary
+	#[pallet::storage]
+	#[allow(clippy::disallowed_types)]
+	/// (ChannelId, PortId, Sequence) => Packet
+	pub type Packets<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
```
Comment (blasrodri) [inline frame/ibc/src/lib.rs:253]: Why not `Blake2128Concat`?

File: scripts/polkadot-launch/composable.json [hunk 4]
```diff
@@ -82,19 +82,19 @@
...
           "basePath": "/tmp/polkadot-launch/parachains/dali-2-bob",
-          "flags": ["--bob", "--rpc-cors=all", "--execution=wasm", "--wasmtime-instantiation-strategy=recreate-instance-copy-on-write"]
+          "flags": ["--bob", "--rpc-cors=all", "--execution=wasm", "--wasmtime-instantiation-strategy=recreate-instance-copy-on-write", "--enable-offchain-indexing=true"]
         },
         {
           "wsPort": 9190,
           "port": 33202,
           "basePath": "/tmp/polkadot-launch/parachains/dali-2-charlie",
-          "flags": ["--charlie", "--rpc-cors=all", "--execution=wasm", "--wasmtime-instantiation-strategy=recreate-instance-copy-on-write"]
+          "flags": ["--charlie", "--rpc-cors=all", "--execution=wasm", "--wasmtime-instantiation-strategy=recreate-instance-copy-on-write", "--enable-offchain-indexing=true"]
```
Comment (blasrodri) [inline scripts/polkadot-launch/composable.json:97]: for those who don't know (like me) what this is: https://docs.substrate.io/main-docs/fundamentals/offchain-operations/

**Review by blasrodri — APPROVED** [review]

File: frame/ibc/src/channel.rs (hunk 2 above)
Comment (seunlanlege) [reply frame/ibc/src/channel.rs:275]: this is a draft pr

File: frame/ibc/src/lib.rs (hunk 3 above)
Comment (seunlanlege) [reply frame/ibc/src/lib.rs:253]: no such thing

File: frame/ibc/ibc-primitives/src/lib.rs (hunk 1 above)
Comment (seunlanlege) [reply frame/ibc/ibc-primitives/src/lib.rs:45]: https://paritytech.github.io/substrate/master/frame_support/pallet_prelude/trait.TypeInfo.html

File: frame/ibc/src/lib.rs (hunk 3 above)
Comment (blasrodri) [reply frame/ibc/src/lib.rs:253]: oh got it, comes from https://docs.rs/frame-support/latest/frame_support/struct.Blake2_128Concat.html
What an odd naming convention

**Pull request merged**